### ISSUE TITLE
start debugger paused in step mode

### ISSUE THREAD

**Issue by microbit-mark** [microsoft/pxt-microbit#3280]
micro:bit support: 38127

**Is your feature request related to a problem? Please describe.**
The debugger starts up in default as "code is running". Using this prpgram  https://makecode.microbit.org/_bDa8J96JAPjq in a very short period of time, the program may have executed hundreds if not thousands of loops.

It would be better if the debugger started up in the pause mode on the very first block of the on start code.

It is seemingly impossible to get this first block, *and* be in the step mode:

By the time the user clicks on the pause button to start the step mode, the program has moved to the next block, even in the snail mode.

**Describe the solution you'd like**
All this would work better by starting with the first block of on start highlighted, and being in the step mode (paused).

**Describe alternatives you've considered**

**Additional context**

**Comment by RRWagner** [comment]
I've recently learned that clicking on the dot of a block in the debugger sets a breakpoint, but there is no reason someone who didn't already know this.. would know to do it.  The breakpoint is a great feature, but it's not self-evident.  I would suggest also adding a help/tip that appears when hovering over the breakpoint dot, as is done in the regular (non-debug mode) editor.

**Comment by abchatra** [comment]
You can restart the execution by clicking restart button and it will hit all breakpoints.
![image](https://user-images.githubusercontent.com/6107272/84854234-f8037780-b015-11ea-9fb5-67c70277f22a.png)

**Comment by RRWagner** [comment]
Yes, I knew about restart.  The issue is that if the user doesn't know that setting the dot equals a breakpoint, what a breakpoint is, and to put a breakpoint on the first block of the program, they "just step through my program".   The alternative is to start the debugger in the step mode, and allow those who wish the program to run full-speed for some amount of time to then click the play button.  That latter approach is user-obvious, knowing what the dot does is less so.

**Comment by abchatra** [comment]
Agreed. I was suggesting a work around today. @riknoll

**Comment by pelikhan** [comment]
Is this the reason we do not get variables on the first run of the debugger? We should restart when entering debugging mode (unless coming from a crash).

**Comment by riknoll** [comment]
@abchatra so are we moving to always start in step mode? iirc that was the original behavior but we changed it because some folks found it confusing.

**Comment by abchatra** [comment]
Let's move back and see. I have heard bunch of people saying they don't understand.

**Comment by riknoll** [comment]
Fixed by https://github.com/microsoft/pxt/pull/8109
